### PR TITLE
fix: extra whitespace view-zone

### PR DESCRIPTION
**Pull request by AhkunTa**
### Types
- [x] 🐛 Bug Fixes

### <samp>🤖 Generated by Copilot at 40a11c6</samp>

*  Filter out comment threads that have no comments to avoid showing empty comment decorations in the editor ([link](https://github.com/opensumi/core/pull/2954/files?diff=unified&w=0#diff-964880a5a3453fde6e41dffaceb5accf42909aea62ca14651f76517ae950127cR484-R490))
*  Simplify the code by removing the redundant condition that checks if the thread has comments ([link](https://github.com/opensumi/core/pull/2954/files?diff=unified&w=0#diff-964880a5a3453fde6e41dffaceb5accf42909aea62ca14651f76517ae950127cL494-L500))
*  Prevent a possible error when the current editor group is not a diff editor by adding an optional chaining operator to access the `diffEditor` property ([link](https://github.com/opensumi/core/pull/2954/files?diff=unified&w=0#diff-964880a5a3453fde6e41dffaceb5accf42909aea62ca14651f76517ae950127cL555-R555))
*  Update comment decorations in diff editors as part of a bug fix ([link](https://github.com/opensumi/core/pull/2954/files?diff=unified&w=0#diff-964880a5a3453fde6e41dffaceb5accf42909aea62ca14651f76517ae950127cL555-R555))

偶现评论内容出现额外的空白
原因是此处 https://github.com/opensumi/core/blob/main/packages/comments/src/browser/comments.service.ts#L217
createThread 会创建冗余数据 

ABC60B2C-4B08-44A5-A4D9-19C7C290FB71

### <samp>🤖 Generated by Copilot at 40a11c6</samp>

Improve comment filtering and decoration in `comments.service.ts`. Fix diff editor bug and hide empty threads.

### CODE REVIEW

**Review by erha19 — APPROVED** [review]

File: packages/comments/src/browser/comments.service.ts [hunk 1]
```diff
@@ -481,6 +481,13 @@ export class CommentsService extends Disposable implements ICommentsService {
       onDidDecorationChange: this.decorationChangeEmitter.event,
       provideEditorDecoration: (uri: URI) =>
         this.commentsThreads
+          .filter((thread) => {
```
Comment (AhkunTa) [inline packages/comments/src/browser/comments.service.ts:484]: 这样直接改有点问题 不然切换的时候可能会导致无法隐藏

**Comment by JimmyVV** [comment]
已经收到哦~~~~

**Review by erha19 — APPROVED** [review]

**Pull request merged**